stable_baselines3/common/save_util.py

```diff
@@ -178,7 +178,9 @@ def json_to_data(json_string: str, custom_objects: Optional[Dict[str, Any]] = No
 
 
 @functools.singledispatch
-def open_path(path: Union[str, pathlib.Path, io.BufferedIOBase], mode: str, verbose: int = 0, suffix: Optional[str] = None):
+def open_path(
+    path: Union[str, pathlib.Path, io.BufferedIOBase], mode: str, verbose: int = 0, suffix: Optional[str] = None
```

Comment:
Why allow `str` paths when you reject them in the assert below?

Reply:
I need to check, might be because of the dispatch. (I didn't write that code)

Reply:
yes, because of the dispatch